lighthouse-core/gather/gather-runner.js

```diff
@@ -96,7 +96,8 @@ class GatherRunner {
       .then(_ => driver.enableRuntimeEvents())
       .then(_ => driver.evaluateScriptOnLoad('window.__nativePromise = Promise;'))
       .then(_ => driver.cleanAndDisableBrowserCaches())
-      .then(_ => driver.clearDataForOrigin(options.url));
+      .then(_ => driver.clearDataForOrigin(options.url))
+      .then(_ => driver.blockUrlPatterns(options.flags.blockedUrlPatterns || []));
```

Comment:
Yeah. As you mentioned, the main reason I did this is because I don't want to add an extra (and optional) parameter in `lighthouse()` and `runLighthouse()`, which looks ugly.